content/en/docs/chart_best_practices/templates.md

```diff
@@ -31,58 +31,81 @@ For that reason, _all defined template names should be namespaced._
 Correct:
 
 ```yaml
-{{- define "nginx.fullname" }}
-{{/* ... */}}
-{{ end -}}
+{{- define "nginx.fullname" -}}
+# ...
+{{- end }}
 ```
 
 Incorrect:
 
 ```yaml
 {{- define "fullname" -}}
-{{/* ... */}}
-{{ end -}}
+# ...
+{{- end }}
 ```
+
 It is highly recommended that new charts are created via `helm create` command
 as the template names are automatically defined as per this best practice.
 
 ## Formatting Templates
 
 Templates should be indented using _two spaces_ (never tabs).
 
-Template directives should have whitespace after the opening  braces and before
+Template directives should have whitespace after the opening braces and before
 the closing braces:
 
 Correct:
-```
-{{ .foo }}
-{{ print "foo" }}
-{{- print "bar" -}}
+
+```yaml
+{{- /* Conditionally create service account */}}
+{{- if .Values.serviceAccount.create }}
+apiVersion: v1
+# ...
+{{- end }}
 ```
 
 Incorrect:
+
+```yaml
+{{-/* This comment is even a syntax error */}}
+{{-if .Values.serviceAccount.create}}
+apiVersion: v1
+# ...
+{{-end}}
 ```
-{{.foo}}
-{{print "foo"}}
-{{-print "bar"-}}
+
+Templates should chomp unnecessary whitespace. A good practice is to
+systematically _chomp left_, but also _chomp right_ for initial content in a
+template file or defined template.
+
+```yaml
+myList:
+  {{- range .Values.elements }}
+  - {{ . }
+  {{- end }}
 ```
 
-Templates should chomp whitespace where possible:
+```yaml
+{{- define "nginx.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "nginx.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+```
 
 ```yaml
-foo:
-  {{- range .Values.items }}
-  {{ . }}
-  {{ end -}}
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+# ...
+{{- end }}
 ```
 
 Blocks (such as control structures) may be indented to indicate flow of the
 template code.
 
-```
-{{ if $foo -}}
-  {{- with .Bar }}Hello{{ end -}}
-{{- end -}}
+```yaml
+{{- range .Values.elements }}
+  {{- print "my element is" . }}
+{{- end }}
```

Comment:
I think the point is useful. The use of indentation.

The old example used both `if` and `with` which is a bad example, too. What about using something like...

```
{{- with .Values.ingress.annotations }}
annotations:
  {{- toYaml . | nindent 4 }}
{{- end }}
```

This is right from generated templates done by `helm create`

Reply:
I think the docs point where this example resides, was orginally to suggest this kind of practice was acceptable, which i personally don't find it to be:

```yaml
annotations:
{{- if something }}
  {{- if something-else }}
    {{- if something-else-again }}
  my.silly.org/annotation: "really silly"
    {{- end }}
  {{- end }}
{{- end }}
```

I think one should always do this instead.

```yaml
annotations:
  {{- if something }}
  {{- if something-else }}
  {{- if something-else-again }}
  my.silly.org/annotation: "really silly"
  {{- end }}
  {{- end }}
  {{- end }}
```

I don't know if there is clear consensus on this kind of practice in the Helm docs anywhere, except for the _helpers.tpl file is mostly following the latter.

Reply:
My recommendation is that this entire example is removed.

> Blocks (such as control structures) may be indented to indicate flow of the
template code.
>
>```yaml
>{{ if $foo -}}
>   {{- with .Bar }}Hello{{ end -}}
>{{- end -}}
>```

Reply:
Or, replaced with an example that one should NOT try to indent templates based on nested template logic, but stay on the indentation level based on the YAML.

DO THIS

```yaml
annotations:
  {{- if something }}
  {{- if something-else }}
  {{- if something-else-again }}
  my.silly.org/annotation: "really silly"
  {{- end }}
  {{- end }}
  {{- end }}
```

DO NOT DO THIS:
```yaml
annotations:
  {{- if something }}
    {{- if something-else }}
      {{- if something-else-again }}
  my.silly.org/annotation: "really silly"
      {{- end }}
    {{- end }}
  {{- end }}
```

Reply:
I conclude that the most reasonable idea is to remove this section. A clear motivation for this is that it is currently having a comment after it saying

Blocks (such as control structures) may be indented to indicate flow of the
template code.

> However, since YAML is a whitespace-oriented language, it is often not possible for code indentation to follow that convention.

See dd2b5ff for the change resolving this discussion in my mind - it removes this section. In my mind, it is a suggestion of a best practice I don't agree with, and, which only is relevant in very rare situations.